alacritty_terminal/src/grid/mod.rs

```diff
@@ -238,21 +239,14 @@ impl<T: GridCell + PartialEq + Copy> Grid<T> {
     }
 
     fn increase_scroll_limit(&mut self, count: usize, template: &T) {
-        self.scroll_limit = min(self.scroll_limit + count, self.max_scroll_limit);
-
-        // Initialize new lines when the history buffer is smaller than the scroll limit
-        let history_size = self.raw.len().saturating_sub(*self.lines);
-        if history_size < self.scroll_limit {
-            let new = min(
-                max(self.scroll_limit - history_size, MIN_INIT_SIZE),
-                self.max_scroll_limit - history_size,
-            );
-            self.raw.initialize(new, Row::new(self.cols, template));
-        }
+        let new = min(count, self.max_scroll_limit - self.history_size());
+        self.raw.initialize(new, Row::new(self.cols, template));
     }
 
     fn decrease_scroll_limit(&mut self, count: usize) {
-        self.scroll_limit = self.scroll_limit.saturating_sub(count);
+        let history_size = self.history_size();
+        let count = history_size - history_size.saturating_sub(count);
```

Comment:
I'd assume the `history_size - history_size.saturating_sub(count)` math is here just to prevent the case where `count > history_size`? Does this actually ever happen?

Otherwise just `count` itself should be sufficient.

Reply:
`scroll_down` unit test was failing without this logic.

Reply:
Might be a good opportunity to validate that the test makes sense. But we can do that when all the other issues have been resolved.

Probably makes sense to resolve the resizing issues first.

Reply:
Well the test makes sense tbh. You shouldn't try to scroll if you reached the end(0).

Reply:
I'm just wondering if there's a more transparent way to do this, since this math seems a bit confusing.

I'd propose using the following instead:

```rust
let count = max(count, self.history_size());
```

I think that's a lot more clear about what we're actually doing, right?